### PR TITLE
1091: Download report (roster and utilization table) as csv

### LINKED ISSUE #990: Update report status
[Mockups](https://app.abstract.com/share/f73a874c-697d-4f0e-bd37-fa167ecc2fa8?collectionId=f17f0f7e-9e6b-45f5-9e83-da84627e951d&collectionLayerId=82156b67-3de1-4adb-85f8-edc7b1589008&present=true&preview=false&sha=c72651e3fe6379138eea392d057a914a75506f7a)

- When I am viewing a report that is not submitted, I see "DRAFT" and the paper icon in the top right.

~~When I am viewing a report editing step (resolving overutilization or missing info), I see "EDITS" and the pencil icon in the top right. Blocked by #982 #983 #984~~

- When I am viewing a report detail page for a submitted report, the top right corner shows "SUBMITTED" and the paper icon with a background color of a light shade of success green (#ECF3EC)

### LINKED ISSUE #1091: I can export my report to CSV
As a provider, I can export my submitted reports to CSV so that I can share and use the data for other purposes, such as sharing with stakeholders and auditors.

When I click on the export icon on the Reports page or when I click on "Export Report" in a report, the data in that report is downloaded to my computer as a CSV. [Designs with export button](https://share.goabstract.com/735b5928-abe5-4e0f-b87a-0b0f7d5b6a24)

The CSV is formatted following this [CDC Export Template](https://docs.google.com/spreadsheets/d/1wuXcoaQnCyJHQomKyECKklDaWER0KWtAJ3115SKyQk4/edit#gid=600449094
).

### PR DESCRIPTION
**Pull request by jlhogan**
should close #1091 

- uses report enrollments for report roster, instead of doing additional roster query
- creates simple util for at least chrome + firefox + hypothetically IE but I haven't tested it yet compat file download (existing js packages did not appear to have all the things we need w/out the things we don't and the code was so small it felt easier to just write it)

also closes #990 
- updates report status to newest specs with icon + tag, replacing the deprecated print button
- puts export report link where report status used to be

PR system link: https://3.17.128.99:5001/

downloaded CSVs:
roster - https://drive.google.com/file/d/17-b5ou5HOHmuZooYXnlCj3Wr3ESyXIuN/view?usp=sharing
revenue - https://drive.google.com/file/d/1DujCHCYyTNmW11eQzuN_NPTHxtCMsLra/view?usp=sharing

### CODE REVIEW

File: src/Hedwig/ClientApp/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ type BaseButtonProps = {
 export type ButtonProps = BaseButtonProps &
 	React.HTMLProps<HTMLButtonElement | HTMLAnchorElement> & {
 		text: string | JSX.Element;
-		onClick?: () => any;
+		onClick?: (_?: any) => any;
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/components/Button/Button.tsx:17]: TODO: port this change to component library

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx [hunk 2]
```diff
@@ -191,3 +254,198 @@ export default function ReportDetail() {
 		</CommonContainer>
 	);
 }
+
+const makeRosterCSVBlob = (enrollments: Enrollment[] | undefined | null) => {
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx:258]: These can probably be moved to utils

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/RevenueView/UtilizationTable.tsx [hunk 3]
```diff
@@ -53,47 +53,145 @@ interface UtilizationTableRow {
 	};
 }
 
+// Not functional component because ts had a problem with the empty fragments
```
Comment (mmazanec22) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/RevenueView/UtilizationTable.tsx:56]: I wonder if using `React.Fragment` instead would help?

**Comment by RyanHansz** [comment]
@jlhogan pull request environment seems down

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx [hunk 4]
```diff
@@ -191,3 +254,198 @@ export default function ReportDetail() {
...
+	enrollments = enrollments || [];
+
+	const fields = [
+		{
+			label: 'Name',
+			value: (row: Enrollment) => nameFormatter(row.child),
+		},
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx:265]: Do we want one name field or separate first and last name fields?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx [hunk 5]
```diff
@@ -191,3 +254,198 @@ export default function ReportDetail() {
...
+	}));
+
+	revenueRows.push(
+		{},
+		{ title: 'Other revenue' },
+		{},
+		{ title: 'Care 4 Kids', total: report.c4KRevenue },
+		{ title: 'Family Fees', total: report.familyFeesRevenue },
+		{},
+		{
```
Comment (cailyncodes) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx:412]: What are the empty `{}` for?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/RevenueView/UtilizationTable.tsx (hunk 3 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/RevenueView/UtilizationTable.tsx:56]: I'm confused by the comment. What's not functional?

Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/RevenueView/UtilizationTable.tsx:56]: this comment existed before i showed up here! it is very possibly out of date, i shall remove it

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx (hunk 4 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx:265]: designs included a template that i matched, which had a single name field

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx (hunk 5 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportDetail.tsx:412]: to add empty rows to match provided template. will comment

**Comment by jlhogan** [comment]
@RyanHansz updated link: https://3.21.227.169:5001

**Comment by RyanHansz** [comment]
Looks good, I'd just change the filename to CDC June 2020 Report - Roster -  Program name.csv

**Pull request merged**